tests/fsharp/core/nameof/preview/test.fsx

```diff
@@ -294,6 +300,30 @@ type Person =
         | x when x = nameof __.Age -> { __ with Age = value :?> int }
         | _ -> __
 
+type GenericClassNameOfTests<'TTT>() =
+
+    static member ``can get name of class type parameter`` () =
+        let b = nameof<'TTT>
+        Assert.AreEqual("TTT", b)
```

Comment:
What about something like this?

```fsharp
type C<'TTT> = class end

nameof<'TTT>
```

Or is that nonsense?

Reply:
`'TTT` is not in scope outside the type declaration here, I wouldn't expect it to work.

Reply:
Yeah, that's true. Makes sense not to add it.